stdlib/3/urllib/request.pyi

```diff
@@ -159,6 +159,11 @@ class ProxyDigestAuthHandler(BaseHandler, AbstractDigestAuthHandler):
 
 class HTTPHandler(BaseHandler):
     def http_open(self, req: Request) -> HTTPResponse: ...
+    def do_open(self,
+                http_class: Callable[  # httplib.client.HTTPConnection protocol
+                    [str, Optional[int], int,
+                     Optional[Tuple[str, int]]], Any],
+                req: Request) -> HTTPResponse: ...
```

Comment:
Since `HTTPConnection` contains optional arguments, it's probably better to use a callable protocol here:

```python
class _HTTPConnection:
    def __call__(self, host: str, port: Optional[int] = ..., ...) -> Any: ...

...

    def do_open(self, http_class: _HTTPConnect, ...)
```

Also, `do_open()` seems to take keyword args.

Reply:
Oh I forgot: The fact that `do_open()` is not documented should be marked with an `# undocumented` comment on the same line.

Reply:
Thanks.  I wasn't sure how best to do this.  A protocol makes sense, though as noted in a comment in the stuff I just pushed I do wonder if HTTPConnection could just be imported from http.client instead of defining our own protocol variant locally?